internal/exec/util/user_lookup.c

```diff
@@ -76,13 +77,39 @@ static int user_lookup_fn(user_lookup_ctxt_t *ctxt) {
 	return 0;
 }
 
+static int group_lookup_fn(lookup_ctxt_t *ctxt) {
```

Comment:
Collapsing `user_lookup_fn` and `group_lookup_fn` would be hard because they're operating on different structs.

What I can do is collapse `user_lookup` and `group_lookup`, because they're almost identical. Just did that.